### PR TITLE
DO NOT MERGE: Testing verify-owners Prow plugin

### PR DESCRIPTION
**Pull request by cjwagner**
/hold
Lets try out the `verify-owners` prow plugin. 🤞 
/cc @matthyx

### CODE REVIEW

**Comment by matthyx** [comment]
Yay!

**Comment by cjwagner** [comment]
```
author:  "cjwagner"   
  component:  "hook"   
  error:  "error creating a review for invalid OWNERS file(s): json: cannot unmarshal string into Go struct field ClientError.errors of type struct { Resource string "json:\"resource\""; Field string "json:\"field\""; Code string "json:\"code\""; Message string "json:\"message,omitempty\"" }"   
  event-GUID:  "19aa3840-764f-11e8-92ee-769b58980bae"   
  event-type:  "pull_request"   
  level:  "error"   
  msg:  "Error handling PullRequestEvent."   
  org:  "kubernetes"   
  plugin:  "verify-owners"   
  pr:  8430   
  repo:  "test-infra"   
  url:  "https://github.com/kubernetes/test-infra/pull/8430"
```

**Comment by matthyx** [comment]
How often is prow redeployed?

**Comment by cjwagner** [comment]
> How often is prow redeployed?

We don't deploy on a schedule yet. @krzyzacy deployed this morning, but had to roll back. Presumably there will be a redeploy once necessary fixes are made.
The `testing-ops` slack channels is where we notify about deploying Prow.

**Comment by cjwagner** [comment]
☹️ Still getting the same error.

**Comment by matthyx** [comment]
Ok, I will investigate...

**Comment by matthyx** [comment]
@cjwagner you can try again once prow has been refreshed :)

**Comment by stevekuznetsov** [comment]
@k8s-ci-robot you are not the boss of me

**Comment by matthyx** [comment]
Lol, @stevekuznetsov is that the secret bypass command?

**Comment by fejta** [comment]
```
  error:  "error creating a review for invalid OWNERS files: json: cannot unmarshal string into Go struct field ClientError.errors of type struct { Resource string "json:\"resource\""; Field string "json:\"field\""; Code string "json:\"code\""; Message string "json:\"message,omitempty\"" }"   
  event-GUID:  "1727fb70-8b9e-11e8-8648-736666d7f632"   
  event-type:  "pull_request"   
  level:  "error"   
  msg:  "Error handling PullRequestEvent."   
```

**Comment by matthyx** [comment]
It looks like the plugin works correctly as in #8782 (minus the review to adjust).
Maybe prow wasn't _yet_ bumped when we did the test...

**Comment by cjwagner** [comment]
@matthyx I think there are still a few problems.
Prow didn't leave a review and threw these errors:
```
{
  author:  "cjwagner"   
  component:  "hook"   
  error:  "couldn't parse hunk on line 5 in patch @@ -3,3 +3,4 @@ approvers:
 - kargakis
 - spxtr
 - stevekuznetsov
+adfasdfjlasdfjlaskdf
\ No newline at end of file: invalid hunk line: \ No newline at end of file"   
  event-GUID:  "a56cf9f0-9ce0-11e8-85fb-eaf053b51166"   
  event-type:  "pull_request"   
  level:  "error"   
  msg:  "Failed to compute added lines in prow/OWNERS: couldn't parse hunk on line 5 in patch @@ -3,3 +3,4 @@ approvers:
 - kargakis
 - spxtr
 - stevekuznetsov
+adfasdfjlasdfjlaskdf
\ No newline at end of file: invalid hunk line: \ No newline at end of file"   
  org:  "kubernetes"   
  plugin:  "verify-owners"   
  pr:  8430   
  repo:  "test-infra"   
  url:  "https://github.com/kubernetes/test-infra/pull/8430"   
 }
```
```
{
  author:  "cjwagner"   
  component:  "hook"   
  error:  "error creating a review for invalid OWNERS files: json: cannot unmarshal string into Go struct field ClientError.errors of type struct { Resource string "json:\"resource\""; Field string "json:\"field\""; Code string "json:\"code\""; Message string "json:\"message,omitempty\"" }"   
  event-GUID:  "a56cf9f0-9ce0-11e8-85fb-eaf053b51166"   
  event-type:  "pull_request"   
  level:  "error"   
  msg:  "Error handling PullRequestEvent."   
  org:  "kubernetes"   
  plugin:  "verify-owners"   
  pr:  8430   
  repo:  "test-infra"   
  url:  "https://github.com/kubernetes/test-infra/pull/8430"   
 }
```

**Comment by matthyx** [comment]
Ok, I will check again... but I swear I saw it working in #8782
And I really need to fix this github client in #8839

File: OWNERS [hunk 1]
```diff
@@ -1,4 +1,4 @@
-approvers:
```
Comment (matthyx) [inline OWNERS]: Cannot parse file: error converting YAML to JSON: yaml: line 6: could not find expected ':'.

Comment (matthyx) [inline OWNERS]: No approvers defined in this root directory OWNERS file.

Comment (matthyx) [inline OWNERS]: No approvers defined in this root directory OWNERS file.

File: mungegithub/OWNERS [hunk 2]
```diff
@@ -8,3 +8,5 @@ reviewers:
 - eparis
```
Comment (matthyx) [inline mungegithub/OWNERS]: Cannot parse file: error converting YAML to JSON: yaml: line 13: could not find expected ':'.

File: prow/OWNERS [hunk 3]
```diff
@@ -3,3 +3,4 @@ approvers:
 - kargakis
```
Comment (matthyx) [inline prow/OWNERS]: Cannot parse file: error converting YAML to JSON: yaml: line 6: could not find expected ':'.

File: OWNERS (hunk 1 above)
Comment (matthyx) [inline OWNERS]: No approvers defined in this root directory OWNERS file.

File: mungegithub/OWNERS (hunk 2 above)
Comment (matthyx) [inline mungegithub/OWNERS]: Cannot parse file: error converting YAML to JSON: yaml: line 13: could not find expected ':'.

File: prow/OWNERS (hunk 3 above)
Comment (matthyx) [inline prow/OWNERS]: Cannot parse file: error converting YAML to JSON: yaml: line 6: could not find expected ':'.

**Comment by matthyx** [comment]
Sorry for the noise @cjwagner but I will be testing with "real" github until I solve this issue.

File: OWNERS (hunk 1 above)
Comment (matthyx) [inline OWNERS]: No approvers defined in this root directory OWNERS file.

File: mungegithub/OWNERS (hunk 2 above)
Comment (matthyx) [inline mungegithub/OWNERS]: Cannot parse file: error converting YAML to JSON: yaml: line 13: could not find expected ':'.

File: prow/OWNERS (hunk 3 above)
Comment (matthyx) [inline prow/OWNERS]: Cannot parse file: error converting YAML to JSON: yaml: line 6: could not find expected ':'.

**Comment by matthyx** [comment]
I hope it is... We could retry once Prow has been bumped with the new error handling, just for the sake of it.
And then if successful, we can enable the plugin for other repos.

**Comment by stevekuznetsov** [comment]
@matthyx we have the bump now

**Comment by cjwagner** [comment]
```
error creating a review for invalid OWNERS files: status code 422 not one of [200], body: {"message":"Validation Failed","errors":["Commit is not part of the pull request"],"documentation_url":"https://developer.github.com/v3/pulls/reviews/#create-a-pull-request-review"}
```
The error message is useful now! The problem is here: https://github.com/kubernetes/test-infra/blob/8a6e78970e96543fb4cdfeda9a82213d3f9e355b/prow/plugins/verify-owners/verify-owners.go#L207-L210

We shouldn't be using the merge commit SHA to create the review. I think that will always fail if the merge commit SHA is available unless the PR is already merged. We should be using `pre.PullRequest.Head.SHA`.

**Comment by cjwagner** [comment]
@matthyx Thanks for your persistence and for adding the alternate error type handling! @stevekuznetsov saw another error using that undocumented format and the GitHub client would have returned a useless message without your change!

**Comment by matthyx** [comment]
Now we can finally enable that plugin on all other projects :)

**Pull request closed without merging**